src/gateway/router/routes/registerRoute.ts

```diff
@@ -0,0 +1,96 @@
+import Router from '@koa/router';
+import { evalType } from '../../tasks/contractsMetadata';
+import { BUNDLR_NODE2_URL } from '../../../constants';
+import { DataItem } from 'arbundles';
+import rawBody from 'raw-body';
+import { Tag } from 'arweave/node/lib/transaction';
+import { sleep } from 'warp-contracts';
+import { updateCache } from '../../updateCache';
+
+export async function registerRoute(ctx: Router.RouterContext) {
+  const { logger, gatewayDb, arweave, bundlr } = ctx;
+
+  const rawDataItem: Buffer = await rawBody(ctx.req);
+  const dataItem = new DataItem(rawDataItem);
+
+  try {
+    const isValid = await dataItem.isValid();
+    if (!isValid) {
+      throw new Error(`Data item binary is not valid.`);
+    }
+
+    const areContractTagsValid = await verifyContractTags(dataItem);
+    if (!areContractTagsValid) {
+      throw new Error(`Contract tags are not valid.`);
+    }
+
+    const bundlrResponse = await bundlr.uploader.transactionUploader(dataItem);
```

Comment:
Don't we check the bundlr response status?
Let's check if we received the full answer (we've got an issue when bundlr replied just 200, but without details and the transactions haven't been bundled)

Reply:
Let's at least check if the necessary fields from bundlr are returned (id, public)

Reply:
you're right. proper check added.

Reply:
another issue here that  - in general - we're not storing in our db the response from bundlr from contracts deployments (only from interactions posted via sequencer).

It shouldn't be _too_ much work to add this to contract deployment (both the old way and the new, from this PR) - especially that probably we will be now spammed with thousands of contracts.